ai_review/algo/pr_processing.py

```
@@ -0,0 +1,550 @@
+    all_models = _get_all_models(model_type)
+    all_deployments = _get_all_deployments(all_models)
+    # try each (model, deployment_id) pair until one is successful, otherwise raise exception
+    for i, (model, deployment_id) in enumerate(zip(all_models, all_deployments)):
+        try:
+            get_logger().debug(
+                f"Generating prediction with {model}"
+                f"{(' from deployment ' + deployment_id) if deployment_id else ''}"
+            )
+            get_settings().set("openai.deployment_id", deployment_id)
+            return await f(model)
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)